lua/autorun/server/sv_acf_caliber_limits.lua

```diff
@@ -0,0 +1,152 @@
+    ply[calibers] = 0
+end )
+
+local getRangeColor
+do
+    local LerpVector = LerpVector
+    local math_Remap = math.Remap
+
+    -- These are vectors so we can LerpVector them
+    local RED = Vector( 1, 0, 0 )
+    local YELLOW = Vector( 1, 1, 0 )
+    local GREEN = Vector( 0, 1, 0 )
+
+    getRangeColor = function( current, limit )
+        local range = current / limit
+
+        local start, finish
+        if range <= 0.5 then
+            range = math_Remap( range, 0, 0.5, 0, 1 )
+            start, finish = GREEN, YELLOW
+        else
+            start, finish = YELLOW, RED
+        end
+
+        return LerpVector( range, start, finish ):ToColor()
+    end
+end
+
+local queueCaliberMessage
+do
+    local timerPrefix = "ACFBalance_CaliberLimitAlert_"
+
+    local function sendCaliberMessage( ply, prefix )
+        prefix = prefix or {}
+
+        if not ply and ply:IsValid() then return end
+
+        -- Just in case one was already queued somehow
+        timer.Remove( timerPrefix .. ply:SteamID64() )
+
+        local current = ply[calibers]
+        local message = "Your current ACF Caliber usage: "
+        local currentCol = getRangeColor( current, caliberLimit )
+
+        local messageTable = table.Add( prefix, {
+            LIGHTGREY, message,
+            currentCol, tostring( current ),
+            LIGHTGREY, " / ",
+            WHITE, tostring( caliberLimit ),
+            LIGHTGREY, " (",
+            currentCol, "∆ ", tostring( caliberLimit - current ),
+            LIGHTGREY, ")\n"
+        } )
+
+        for gun, caliber in pairs( ply[guns] ) do
+            table.Add( messageTable, {
+                LIGHTGREY, "  - ",
+                WHITE, gun.Name,
+                LIGHTGREY, " (", gun:GetClass(), "): ",
+                getRangeColor( caliber, maxCaliber ), tostring( caliber ),
+                "\n"
+            } )
+        end
+
+        net.Start( "ACFBalance_CaliberLimitAlert" )
+        net.WriteTable( messageTable )
+        net.Send( ply )
+    end
+
+    queueCaliberMessage = function( ply, prefix, delay )
+        delay = delay or 0.15
+
+        timer.Create( timerPrefix .. ply:SteamID64(), delay, 1, function()
+            sendCaliberMessage( ply, prefix )
+        end )
+    end
+
+    hook.Add( "PlayerSay", "ACFBalance_CaliberLimit", function( ply, text )
+        if text ~= "!calibers" then return end
+
+        queueCaliberMessage( ply )
+        return ""
+    end )
+end
+
+do
+    local hasCalibers = {
+        acf_gun = true,
+        acf_missile = true
+    }
+
+    hook.Add( "ACF_OnEntitySpawn", "ACFBalance_CaliberLimit", function( what, ent, entData )
+        if not hasCalibers[what] then return end
+
+        local ply = ent.Owner
+        local plyTable = ply:GetTable()
+        local gunCaliber = getCaliber( what, entData )
+
+        plyTable[guns][ent] = gunCaliber
+        ent:CallOnRemove( "ACFBalance_CaliberLimit", function()
+            if not ply and ply:IsValid() then return end
```

Comment:
Fix logic.
```suggestion
            if not ply or not ply:IsValid() then return end
```
Could also drop the `not ply or` if a validity check is done prior, ensuring that it's non-nil.

Reply:
Gotta be done here, their props could be removed after they leave